bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -71,32 +71,20 @@
             .of(THING_TYPE_SOCKET, THING_TYPE_INSIGHT, THING_TYPE_LIGHTSWITCH, THING_TYPE_MOTION)
             .collect(Collectors.toSet());
 
-    private Map<String, Boolean> subscriptionState = new HashMap<>();
+    private final UpnpIOService service;
+
+    private final Object upnpLock = new Object();
+    private final Object jobLock = new Object();
 
     private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
 
-    protected UpnpIOService service;
     private WemoHttpCall wemoCall;
 
-    private @Nullable ScheduledFuture<?> refreshJob;
+    private Map<String, Boolean> subscriptionState = new HashMap<>();
 
-    private final Runnable refreshRunnable = new Runnable() {
+    private @Nullable ScheduledFuture<?> pollingJob;
 
-        @Override
-        public void run() {
-            try {
-                if (!isUpnpDeviceRegistered()) {
-                    logger.debug("WeMo UPnP device {} not yet registered", getUDN());
-                }
-
-                updateWemoState();
-                onSubscription();
-            } catch (Exception e) {
-                logger.debug("Exception during poll", e);
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            }
-        }
-    };
+    private String host = "";
```

Comment:
Please decide if missing host should be null or empty as currently it's a combination and null annotation warnings have been introduced. If you stick with empty string as missing host (hence `@NonNull`) you can skip the "host == null" checks:
```
[WARNING] openhab-addons\bundles\org.openhab.binding.wemo\src\main\java\org\openhab\binding\wemo\internal\handler\WemoHandler.java:[139,21] Redundant null check: comparing '@NonNull String' against null
[WARNING] openhab-addons\bundles\org.openhab.binding.wemo\src\main\java\org\openhab\binding\wemo\internal\handler\WemoHandler.java:[169,13] Redundant null check: comparing '@NonNull String' against null
[WARNING] openhab-addons\bundles\org.openhab.binding.wemo\src\main\java\org\openhab\binding\wemo\internal\handler\WemoHandler.java:[414,13] Redundant null check: comparing '@NonNull String' against null
```

Reply:
OK, will remove the null check later on...

Reply:
Removed redundant null checks